File: 2015/Day6/fireHazard.py
import re
 
lightMatrix = [[0 for x in range(1000)] for x in range(1000)] 
brightnessCounter = 0
 
with open('input.txt') as f:
  for line in f:
    # Match command
    res = re.search(r'(turn on|turn off|toggle)\s*(\d*?),(\d*?)\s*through\s(\d*?),(\d*)', line)
    command, lower_X, lower_Y, upper_X, upper_Y =  res.group(1), int(res.group(2)), int(res.group(3)), int(res.group(4)), int(res.group(5))
    
    # Perform actions
    if command == 'turn off':
      data = -1
    elif command == 'turn on':
      data = 1
    elif command == 'toggle':
      data = 2
      
    # Apply action
    for x_val in range (lower_X, upper_X + 1):
      for y_val in range (lower_Y, upper_Y + 1):
        lightMatrix[x_val][y_val] += data if lightMatrix[x_val][y_val] + data >= 0 else 0
        
# Get brightness level
for i in lightMatrix:
  for brightness in i:
    brightnessCounter += brightness
 
# answer
print(brightnessCounter)

File: 2015/Day6/fireHazard.R
#!/usr/bin/Rscript

#clear workspace
rm(list = ls())

# Libraries

#------------------------------------------------------------------------------------------
#Load Data
#------------------------------------------------------------------------------------------

#Use this if you're running from the command line:
# mydata <- readLines(file("stdin"))

#Use this if you're opening this repo as a R project, using relative paths:
# mydata <- readLines('2015/Day6/input.txt')
mydata <- readLines('2015/Day6/input.txt')

#------------------------------------------------------------------------------------------
#Part 1
#------------------------------------------------------------------------------------------
df <- setNames(data.frame(matrix(ncol = 5, nrow = length(mydata))), 
                   c("command","lower_x","lower_y","upper_x","upper_y"))

for(i in 1:length(mydata)){
  mystring <- strsplit(mydata[i]," ")[[1]]
  if(mystring[1]=="turn"){
    df$command[i] <- paste(mystring[1],mystring[2],sep=" ")
    
    df$lower_x[i] <- as.numeric(strsplit(mystring[3],",")[[1]][1])
    df$lower_y[i] <- as.numeric(strsplit(mystring[3],",")[[1]][2])
    
    df$upper_x[i] <- as.numeric(strsplit(mystring[5],",")[[1]][1])
    df$upper_y[i] <- as.numeric(strsplit(mystring[5],",")[[1]][2])
  }
  
  
  if(mystring[1]=="toggle"){
    df$command[i] <- "toggle"
    
    df$lower_x[i] <- as.numeric(strsplit(mystring[2],",")[[1]][1])
    df$lower_y[i] <- as.numeric(strsplit(mystring[2],",")[[1]][2])
    
    df$upper_x[i] <- as.numeric(strsplit(mystring[4],",")[[1]][1])
    df$upper_y[i] <- as.numeric(strsplit(mystring[4],",")[[1]][2])
  }
  
}


lightMatrix <- matrix(FALSE,ncol = 1000, nrow = 1000)
toggle <- FALSE
data_ <- FALSE

for(i in 1:nrow(df)){
  if(df$command[i]=="turn off"){
    data_ <- FALSE
  }else if(df$command[i]=="turn on"){
    data_ <- TRUE
  }else if(df$command[i]=="toggle"){
    toggle <- TRUE
  }
  
  for (x_val in (df$lower_x[i]):(df$upper_x[i])) {
    for (y_val in (df$lower_y[i]):(df$upper_y[i])){
      if(toggle){
        lightMatrix[x_val,y_val] <- !lightMatrix[x_val,y_val]
      }else{
        lightMatrix[x_val,y_val] <- data_
      }
    }
  }
}

light_counter <- sum(lightMatrix)
light_counter
# 569999
#not getting the right answer
